lib/bosh/BoshDirectorClient.js

```diff
@@ -432,7 +434,16 @@ class BoshDirectorClient extends HttpClient {
         }
       })
       .then(manifest => {
-        return manifest.properties.agent;
+        if (manifest.instance_groups) {
+          let agentJob = {};
+          _.each(manifest.instance_groups, (instance_group) => {
+            agentJob = _.find(instance_group.jobs, job => job.name === 'broker-agent');
+            return !agentJob;
+          });
+          return agentJob.properties;
+        } else {
+          return manifest.properties.agent;
```

Comment:
put a comment here that this is to handle backward compatibility and a TODO: to remove this once all instances are migrated. Else down (say 3 months) down the line this becomes confusing.